src/gui/qgsmapcanvas.h

```diff
@@ -212,6 +212,23 @@ class GUI_EXPORT QgsMapCanvas : public QGraphicsView
      */
     void setMapUpdateInterval( int timeMilliseconds );
 
+    /**
+     * Populates the QgsMapCanvas context \a menu prior to showing it in response to a right-mouse button click.
+     *
+     * The \a event allows to consider the mouse-pixel position, e.g. to account for
+     * its map-coordinates.
+     *
+     * This method is called after the \a menu is populated by the active map tool
+     *
+     * - 
+     *
+     * \note The context menu is only shown when the ShowContextMenu flag
+     * is present in flags().
+     *
+     * \since QGIS 3.16
+     */
+    void populateContextMenu( QMenu *menu, QgsMapMouseEvent *event );
```

Comment:
This will need to be virtual, or the redirection in `showEvent` will not work.

Reply:
I gather from the linked ticket that the intention was for this to be a signal that plugins could connect to and manipulate the menu

Reply:
That sounds like a very good approach to move forward :+1:

Reply:
Ok, then I'm going to make ``populateContextMenu`` a signal.